### PR TITLE
hiworkload: check missing dependencies

### PR DESCRIPTION
**Pull request by pevik**
Basically print "TEST SKIPPED" instead of "TEST FAILED" when missing dependencies to compile `hiworkload`.

### CODE REVIEW

File: hiworkload/compile.sh [hunk 1]
```diff
@@ -1,8 +1,20 @@
 #!/bin/sh
 # SPDX-License-Identifier: GPL-2.0-or-later
-# Copyright (c) 2019 Petr Vorel <pvorel@suse.cz>
+# Copyright (c) 2019-2022 Petr Vorel <pvorel@suse.cz>
```
Comment (mirab) [inline hiworkload/compile.sh]: A nit but this is generally not required and 2019 would still be fine here. JFYI.

File: hiworkload/compile.sh [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+missing=
+for cmd in aclocal autoconf autoheader automake make; do
+	if ! command -v $cmd >/dev/null; then
+		echo "Missing '$cmd', install it!"
+		missing=1
+	fi
+done
+
+if [ "$missing" ]; then
+	exit 99
```
Comment (mirab) [inline hiworkload/compile.sh]: There is exit code 22 used for skipped elsewhere in the code. Do we want to stick to that convention?

File: klp_tc_functions.sh [hunk 3]
```diff
@@ -288,7 +288,7 @@ function klp_tc_cleanup() {
 }
 
 function klp_tc_init() {
-    trap "[ \$? -ne 0 ] && call_recovery_hooks; klp_tc_cleanup; echo TEST FAILED while executing \'\$BASH_COMMAND\'" EXIT
+    trap "ret=\$?; [ \$ret -ne 0 ] && call_recovery_hooks; klp_tc_cleanup; result=FAILED; [ \$ret -ne 99 ] && result=SKIPPED; echo TEST \$result while executing \'\$BASH_COMMAND\'" EXIT
```
Comment (mirab) [inline klp_tc_functions.sh]: I may misread this, but shouldn't there be

[ \$ret -eq 99 ] && result=SKIPPED;

?

Comment (pevik) [reply klp_tc_functions.sh]: Good catch. Also I had to add one more place where error code evaluation was needed.

**Comment by pevik** [comment]
Hopefully everything fixed. Feel free to adjust it whatever you want before merge.

**Comment by mirab** [comment]
Merged

**Pull request closed without merging**